legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/RelationalLSPGrammarExtension.java

```diff
@@ -52,6 +55,89 @@ public class RelationalLSPGrammarExtension extends AbstractSectionParserLSPGramm
+            "TableName1(column1 INT PRIMARY KEY, column2 DATE)\n"
+    );
+
+    private static final ImmutableList<String> VIEW_TRIGGERS = Lists.immutable.with("View ");
+
+    private static final ImmutableList<String> VIEW_SUGGESTIONS = Lists.immutable.with(
+            "viewName\n" +
+                "(\n" +
+                "  field1: table1.column1, \n" +
+                "  field2: table2.column2,\n" +
+                "  field3: table1.column1 + table2.column2\n" +
+                ")\n"
+    );
+
+    private static final ImmutableList<String> JOIN_TRIGGERS = Lists.immutable.with("Join ");
+
+    private static final ImmutableList<String> JOIN_SUGGESTIONS = Lists.immutable.with(
+            "joinName(table1.column1 = table2.column2)"
+    );
+
+    private static final ImmutableList<String> FILTER_TRIGGERS = Lists.immutable.with("Filter ");
+
+    private static final ImmutableList<String> FILTER_SUGGESTIONS = Lists.immutable.with(
+            "filterName(table1.column1 > 12)"
+    );
+
+    private static final ImmutableList<String> BOILERPLATE_SUGGESTIONS = Lists.immutable.with(
+            "Database package::path::storeName\n" +
+            "(\n" +
+            "  Schema schemaName\n" +
+            "  (\n" +
+            "   Table TableName1(column1 INT PRIMARY KEY, column2 DATE)\n" +
+            "   Table TableName2(column3 VARCHAR(10) PRIMARY KEY)\n" +
+            "  )\n" +
+            ")\n"
+    );
+
+    public Iterable<? extends LegendCompletion> getCompletions(SectionState section, TextPosition location)
+    {
+        String codeLine = section.getSection().getLine(location.getLine()).substring(0, location.getColumn());
+        List<LegendCompletion> legendCompletions = Lists.mutable.empty();
+
+        if (codeLine.isEmpty())
+        {
+            return BOILERPLATE_SUGGESTIONS.collect(s -> new LegendCompletion("Relational boilerplate", s.replaceAll("\n",System.getProperty("line.separator"))));
+        }
+
+        if (SCHEMA_TRIGGERS.anySatisfy(codeLine::endsWith))
+        {
+            SCHEMA_SUGGESTIONS.collect(s -> new LegendCompletion("Schema definition", s), legendCompletions);
+        }
+        if (TABLE_TRIGGERS.anySatisfy(codeLine::endsWith))
+        {
+            TABLE_SUGGESTIONS.collect(s -> new LegendCompletion("Table definition", s), legendCompletions);
+        }
+        if (VIEW_TRIGGERS.anySatisfy(codeLine::endsWith))
+        {
+            VIEW_SUGGESTIONS.collect(s -> new LegendCompletion("View definition", s), legendCompletions);
+        }
+        if (JOIN_TRIGGERS.anySatisfy(codeLine::endsWith))
+        {
+            JOIN_SUGGESTIONS.collect(s -> new LegendCompletion("Join definition", s), legendCompletions);
+        }
+        if (FILTER_TRIGGERS.anySatisfy(codeLine::endsWith))
+        {
+            FILTER_SUGGESTIONS.collect(s -> new LegendCompletion("Filter definition", s), legendCompletions);
+        }
```

Comment:
I believe all of these should have their line separators replaced. Perhaps you should abstract out a function that does it?